service/worker/scanner/executions/scavenger.go

```diff
@@ -154,12 +153,24 @@ func (s *Scavenger) Alive() bool {
 
 // run does a single run over all executions and validates them
 func (s *Scavenger) run() {
-	// TODO: implement this
-	// 1. read from visibility records from frontend.Client
-	// 2. create executionTasks
-	// 3. pass them off to the executor to run
-	// 4. wait until the executor is done
-	// 5. emit metrics on the run of scavenger
+	defer func() {
+		go s.Stop()
+		s.stopWG.Done()
+	}()
+
+	rateLimiter := getRateLimiter(targetRPS)
+	if rateLimiter == nil {
+		s.logger.Error("ExecutionsScavenger failed to initialize rate limiter. Aborting scan.")
+		return
+	}
```

Comment:
just panic?

Reply:
If `Scavenger` should never start w/o rate limiter being set, I agree to `panic` here. Message should explicitly say it though. Something like: "Rate limiter is not configured in Scavenger config. Disable Scavenger or add rate limiter config."